### ISSUE TITLE
[BUG]: requirements.txt has duplicate `plotly` entry and no version pinning — causes environment inconsistency

### ISSUE THREAD

**Issue by Tech4Aditya** [aliviahossain/Disease-prediction#333]
## Describe the Bug

`requirements.txt` has two concrete problems that will silently break fresh installs and CI environments:

### 1. Duplicate `plotly` dependency (lines 15 and 22)

```txt
...
plotly   ← line 15
...
plotly   ← line 22 (duplicate)
```

While `pip` won't crash on this, it installs the package twice during resolution, pollutes the dependency graph, and makes auditing harder.

### 2. Zero version pinning across all 22 dependencies

Not a single package has a pinned version. This means:
- `pip install -r requirements.txt` produces **different environments on different machines/days**
- Breaking changes in upstream packages (e.g. `google-genai`, `scikit-learn`, `tensorflow-cpu`) will silently break the app with no warning
- Contributors and CI runs can't reproduce identical environments

## Steps to Reproduce

```bash
pip install -r requirements.txt
pip list | grep plotly
# shows plotly installed, but requirements has it twice
```

## Expected Behavior

- `plotly` should appear **once**
- All packages should have pinned or at minimum minimum-bounded versions, e.g. `flask>=3.0,<4.0`

## Suggested Fix

1. Remove the duplicate `plotly` on line 22
2. Pin versions using `pip freeze > requirements.txt` after a clean working install, or at minimum add lower bounds like:
```txt
flask>=3.0
scikit-learn>=1.4
plotly>=5.0
tensorflow-cpu>=2.15
```

## Environment

- File: `requirements.txt`
- Affects: all contributors doing fresh clone + install

## Labels

`bug`, `dependencies`, `GSSoC`
